packages/vite/src/node/plugins/resolve.ts

```diff
@@ -555,10 +552,21 @@ function tryCleanFsResolve(
     if (dirStat?.isDirectory()) {
       if (possibleJsToTs) {
         // try resolve .js, .mjs, .mts or .jsx import to typescript file
-        const tsSrcPaths = getPotentialTsSrcPaths(file)
-        for (const srcPath of tsSrcPaths) {
-          if ((res = tryResolveRealFile(srcPath, preserveSymlinks))) return res
-        }
+        const type = path.extname(file)
+        const fileName = file.slice(0, -type.length)
+        if (
+          (res = tryResolveRealFile(
+            fileName + type.replace('js', 'ts'),
+            preserveSymlinks,
+          ))
+        )
+          return res
+        // for .js, also try .tsx
+        if (
+          type === '.js' &&
+          (res = tryResolveRealFile(fileName + '.tsx', preserveSymlinks))
+        )
+          return res
```

Comment:
`getPotentialSrcPaths` was returning the following:
- .js -> [.ts, .tsx]
- .mjs -> [.mts, .mtsx] 👀
- .cjs -> [.cts, .ctsx] 👁️ 👁️ 
- .jsx -> [.tsx]

It was also doing a complex regex taking into account query params. This function was hiding the real checks, better to be explicit and have them expanded here.
Now we no longer check for `mtsx` and `ctsx`